elastic/eui
# [EuiDataGrid] Fix cell expansion popover not correctly closing when clicking the originating cell

### Summary

closes https://github.com/elastic/eui/issues/5190

The main fix is the `onTrapDeactivation={closeCellPopover}` prop addition, the other diffs are me just reordering props

## QA

![screencap](https://user-images.githubusercontent.com/549407/156459536-019447e7-d247-4b04-ac2c-c786492839e9.gif)

- Go to http://localhost:8030/#/tabular-content/data-grid-row-heights-options#auto-heights-for-rows
- Open any cell popover
- [x] Confirm that clicking back into the cell that triggered the popover closes the popover
- [x] Confirm that clicking anywhere outside the cell (e.g. another cell, anywhere on the page) also correctly closes the popover

### Checklist

~- [ ] Check against **all themes** for compatibility in both light and dark modes~
~- [ ] Checked in **mobile**~
~- [ ] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**~
~- [ ] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#adding-playground-toggles)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md)**~
~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
~- [ ] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/cypress-testing.md) tests**~
~- [ ] Checked for **breaking changes** and labeled appropriately~
